clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```diff
@@ -664,6 +674,8 @@ public void close(Duration timeout) {
         long now = time.milliseconds();
         long newHardShutdownTimeMs = now + waitTimeMs;
         long prev = INVALID_SHUTDOWN_TIME;
+        clientTelemetryReporter.ifPresent(ClientTelemetryReporter::initiateClose);
+        metrics.close();
```

Comment:
Just double checking: could either `initiateClose` or `metrics.close()` throw? If yes, would it be a problem that we don't execute the code below?

Reply:
No, neither method throws.